TidepoolKit/Internal/DNS.swift

```diff
@@ -17,11 +17,32 @@ enum DNSError: Error {
 struct DNSSRVRecord: Equatable {
     let priority: UInt16
     let weight: UInt16
-    let port: UInt16
     let host: String
+    let port: UInt16
+}
+
+extension DNSSRVRecord: Comparable {
+    public static func < (lhs: DNSSRVRecord, rhs: DNSSRVRecord) -> Bool {
+        if lhs.priority < rhs.priority {
+            return true
+        } else if lhs.priority > rhs.priority {
+            return false
+        }
+        if lhs.weight > rhs.weight {
+            return true
+        } else if lhs.weight < rhs.weight {
+            return false
+        }
+        if lhs.host < rhs.host {
+            return true
+        } else if lhs.host > rhs.host {
+            return false
+        }
```

Comment:
[nit] This portion of the comparison is confusing to me as I don't understand why the `host` construction is part of this comparison. The previous comparison where equal hosts use the port and `localhost` is being detected was easier for me to understand.

Reply:
This comparison is only used to order the environments in a rationale and stable manner. It uses priority and weight from the SRV record first. Then, if those are identical for two environments, it alphabetically compares the hosts. Then, if the hosts match, compares ports. 

I moved the check for `localhost` outside of this comparison since it is overriding the natural order to "fix" something the backend should change (in my opinion). I recommended to the backend that they update the DNS SRV records to actually order via priority and weight so *all* clients get the same rational ordering (production, integration, qa1, and qa2) and `localhost` would only be added manually by the clients if desired (not in the DNS SRV record since that should only be used to advertise environments hosted by the backed; `localhost` is hosted by the local development environment and shouldn't be seen by normal users even if they know the special way to switch environments).

Reply:
So, with that said, going to leave this as-is.